### PR TITLE
Don't use nbdkit for xz or gz

### LINKED ISSUE #2288: fcos qcow2.xz fails to decompress and import
**What happened**:

Receiving an error when trying to import https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/35.20220424.3.0/x86_64/fedora-coreos-35.20220424.3.0-qemu.x86_64.qcow2.xz

**What you expected to happen**:

Success!

**How to reproduce it (as minimally and precisely as possible)**:

```yaml
---
apiVersion: cdi.kubevirt.io/v1beta1
kind: DataVolume
metadata:
  name: fedora-coreos-35-x86-64
spec:
  source:
    http:
      url: https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/35.20220424.3.0/x86_64/fedora-coreos-35.20220424.3.0-qemu.x86_64.qcow2.xz
  pvc:
    accessModes:
      - ReadWriteOnce
    resources:
      requests:
        storage: 35G
```

Below is an error while importing centos-7 image. I don't care about this image or the
nbdkit error, just demonstrating that it tries to decompress the image. Importing https://mirror.arizona.edu/fedora/linux/releases/34/Cloud/x86_64/images/Fedora-Cloud-Base-34-1.2.x86_64.raw.xz also works fine.

```yaml
---
apiVersion: cdi.kubevirt.io/v1beta1
kind: DataVolume
metadata:
  name: centos-7-x86-64
spec:
  source:
    http:
      url: https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2.xz
  pvc:
    accessModes:
      - ReadWriteOnce
    resources:
      requests:
        storage: 35G
```

Logs:

```shell
❯ k logs importer-centos-7-x86-64 -n kargo
I0522 16:46:51.974049       1 importer.go:83] Starting importer
I0522 16:46:51.974128       1 importer.go:150] begin import process
I0522 16:46:52.564636       1 data-processor.go:379] Calculating available size
I0522 16:46:52.564779       1 data-processor.go:391] Checking out file system volume size.
I0522 16:46:52.564797       1 data-processor.go:399] Request image size not empty.
I0522 16:46:52.564807       1 data-processor.go:404] Target size 35G.
I0522 16:46:52.806885       1 nbdkit.go:294] Waiting for nbdkit PID.
I0522 16:46:53.307895       1 nbdkit.go:315] nbdkit ready.
I0522 16:46:53.307982       1 data-processor.go:282] New phase: Convert
I0522 16:46:53.308009       1 data-processor.go:288] Validating image
I0522 16:46:53.790330       1 nbdkit.go:281] Log line from nbdkit: nbdkit: curl[1]: error: xz file largest block is bigger than maxblock
I0522 16:46:53.790517       1 nbdkit.go:281] Log line from nbdkit: Either recompress the xz file with smaller blocks (see nbdkit-xz-filter(1))
I0522 16:46:53.790579       1 nbdkit.go:281] Log line from nbdkit: or make maxblock parameter bigger.
I0522 16:46:53.790596       1 nbdkit.go:281] Log line from nbdkit: maxblock = 536870912 (bytes)
I0522 16:46:53.790606       1 nbdkit.go:281] Log line from nbdkit: largest block in xz file = 858783744 (bytes)
E0522 16:46:53.792399       1 prlimit.go:174] qemu-img failed output is:
E0522 16:46:53.792411       1 prlimit.go:175]
E0522 16:46:53.792417       1 prlimit.go:176] qemu-img: Could not open 'nbd+unix:///?socket=/var/run/nbdkit.sock': Requested export not available

E0522 16:46:53.792481       1 data-processor.go:278] qemu-img: Could not open 'nbd+unix:///?socket=/var/run/nbdkit.sock': Requested export not available
, qemu-img execution failed: exit status 1 Log line from nbdkit: nbdkit: curl[1]: error: xz file largest block is bigger than maxblockLog line from nbdkit: Either recompress the xz file with smaller blocks (see nbdkit-xz-filter(1))Log line from nbdkit: or make maxblock parameter bigger.Log line from nbdkit: maxblock = 536870912 (bytes)Log line from nbdkit: largest block in xz file = 858783744 (bytes)
Unable to convert source data to target format
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:242
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
E0522 16:46:53.792542       1 importer.go:159] qemu-img: Could not open 'nbd+unix:///?socket=/var/run/nbdkit.sock': Requested export not available
, qemu-img execution failed: exit status 1 Log line from nbdkit: nbdkit: curl[1]: error: xz file largest block is bigger than maxblockLog line from nbdkit: Either recompress the xz file with smaller blocks (see nbdkit-xz-filter(1))Log line from nbdkit: or make maxblock parameter bigger.Log line from nbdkit: maxblock = 536870912 (bytes)Log line from nbdkit: largest block in xz file = 858783744 (bytes)
Unable to convert source data to target format
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:242
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
```

Error when importing fedora-coreos-35 shows that there is an error while trying to convert the image but it doesn't seem to be getting decrompressed at all.

```
❯ k logs importer-fedora-coreos-35-x86-64 -n kargo
I0522 16:50:30.874951       1 importer.go:83] Starting importer
I0522 16:50:30.875044       1 importer.go:150] begin import process
I0522 16:50:31.223145       1 data-processor.go:379] Calculating available size
I0522 16:50:31.223171       1 data-processor.go:391] Checking out file system volume size.
I0522 16:50:31.223181       1 data-processor.go:399] Request image size not empty.
I0522 16:50:31.223188       1 data-processor.go:404] Target size 35G.
I0522 16:50:31.250787       1 nbdkit.go:294] Waiting for nbdkit PID.
I0522 16:50:31.751103       1 nbdkit.go:315] nbdkit ready.
I0522 16:50:31.751255       1 data-processor.go:282] New phase: Convert
I0522 16:50:31.751276       1 data-processor.go:288] Validating image
I0522 16:50:37.667032       1 qemu.go:259] 0.00
I0522 16:50:43.902369       1 qemu.go:259] 1.04
I0522 16:50:53.750154       1 qemu.go:259] 2.14
I0522 16:50:53.771191       1 qemu.go:259] 3.18
I0522 16:50:53.782090       1 qemu.go:259] 4.20
I0522 16:50:53.791181       1 qemu.go:259] 5.23
I0522 16:50:53.866528       1 qemu.go:259] 6.26
I0522 16:50:56.000970       1 nbdkit.go:287] Stopped watching nbdkit log.
E0522 16:50:56.003831       1 prlimit.go:174] qemu-img failed output is:
E0522 16:50:56.003841       1 prlimit.go:175]     (0.00/100%)
    (1.04/100%)
    (2.14/100%)
    (3.18/100%)
    (4.20/100%)
    (5.23/100%)
    (6.26/100%)


E0522 16:50:56.003847       1 prlimit.go:176] qemu-img: error while reading at byte 538050560: Input/output error
qemu-img: error while reading at byte 538116096: Input/output error
qemu-img: error while reading at byte 538181632: Input/output error
qemu-img: error while reading at byte 538247168: Input/output error
qemu-img: error while reading at byte 538312704: Input/output error
qemu-img: error while reading at byte 538378240: Input/output error
qemu-img: error while reading at byte 538443776: Input/output error
qemu-img: error while reading at byte 538509312: Input/output error

E0522 16:50:56.005723       1 data-processor.go:278] exit status 1
qemu-img execution failed
kubevirt.io/containerized-data-importer/pkg/system.executeWithLimits
	pkg/system/prlimit.go:178
kubevirt.io/containerized-data-importer/pkg/system.ExecWithLimits
	pkg/system/prlimit.go:111
kubevirt.io/containerized-data-importer/pkg/image.convertToRaw
	pkg/image/qemu.go:128
kubevirt.io/containerized-data-importer/pkg/image.(*qemuOperations).ConvertToRawStream
	pkg/image/qemu.go:146
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).convert
	pkg/importer/data-processor.go:303
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:240
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
could not convert image to raw
kubevirt.io/containerized-data-importer/pkg/image.convertToRaw
	pkg/image/qemu.go:136
kubevirt.io/containerized-data-importer/pkg/image.(*qemuOperations).ConvertToRawStream
	pkg/image/qemu.go:146
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).convert
	pkg/importer/data-processor.go:303
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:240
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
Conversion to Raw failed
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).convert
	pkg/importer/data-processor.go:305
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:240
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
Unable to convert source data to target format
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:242
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
E0522 16:50:56.005823       1 importer.go:159] exit status 1
qemu-img execution failed
kubevirt.io/containerized-data-importer/pkg/system.executeWithLimits
	pkg/system/prlimit.go:178
kubevirt.io/containerized-data-importer/pkg/system.ExecWithLimits
	pkg/system/prlimit.go:111
kubevirt.io/containerized-data-importer/pkg/image.convertToRaw
	pkg/image/qemu.go:128
kubevirt.io/containerized-data-importer/pkg/image.(*qemuOperations).ConvertToRawStream
	pkg/image/qemu.go:146
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).convert
	pkg/importer/data-processor.go:303
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:240
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
could not convert image to raw
kubevirt.io/containerized-data-importer/pkg/image.convertToRaw
	pkg/image/qemu.go:136
kubevirt.io/containerized-data-importer/pkg/image.(*qemuOperations).ConvertToRawStream
	pkg/image/qemu.go:146
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).convert
	pkg/importer/data-processor.go:303
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:240
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
Conversion to Raw failed
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).convert
	pkg/importer/data-processor.go:305
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:240
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
Unable to convert source data to target format
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func6
	pkg/importer/data-processor.go:242
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
	pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
	pkg/importer/data-processor.go:184
main.handleImport
	cmd/cdi-importer/importer.go:156
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:255
runtime.goexit
	GOROOT/src/runtime/asm_amd64.s:1581
```

**Additional context**:

It seems that the image is never being unarchived like the centos image is then
erroring when trying to convert to raw with `qemu-img`.

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`): `v1.49.0`
- Kubernetes version (use `kubectl version`): 1.22

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

Nbdkit is not a good fit for gz because it downloads/unzips the file to tempdir.

It is also not a good fit for xz, as discussed in #2288.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2288

https://bugzilla.redhat.com/show_bug.cgi?id=2089391

**Special notes for your reviewer**:

**Release note**:

```release-note
NBDKit no longer used for .gz or .xz files
```

### CODE REVIEW

**Comment by awels** [comment]
Seems like the same tests failed again? Try reproducing locally?

File: pkg/image/filefmt.go [hunk 1]
```diff
@@ -54,13 +54,14 @@ var knownHeaders = Headers{
 	},
 	"vdi": Header{
 		Format:      "vdi",
-		magicNumber: []byte("<<< Oracle VM"),
+		magicNumber: []byte{0x7F, 0x10, 0xDA, 0xBE},
+		mgOffset:    0x40,
```
Comment (mhenriks) [inline pkg/image/filefmt.go:58]: these fixes should probably be backported

Comment (mhenriks) [reply pkg/image/filefmt.go:58]: Upon further though, probably not necessary because `qemu-img convert` is still called

**Review by akalenyu — COMMENTED** [review]
Looks good, might make sense to have this doc reflect these changes
https://github.com/kubevirt/containerized-data-importer/blob/d9f4ff21a4532024bd06c1e016e73b9465cd76e9/doc/scratch-space.md

**Review by awels — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**